tokio-stream/src/wrappers/stream_close.rs

```diff
@@ -0,0 +1,59 @@
+use crate::Stream;
+use pin_project_lite::pin_project;
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+pin_project! {
+    /// Stream that returns None when the stream is exhausted. To be used with StreamMap to know when a stream ends.
+    #[must_use = "streams do nothing unless polled"]
+    pub struct StreamNotifyClose<S> {
+        #[pin]
+        inner: S,
+        ended: bool
+    }
+}
+
+impl<S> StreamNotifyClose<S> {
+    /// Create a new `StreamNotifyClose`.
+    pub fn new(stream: S) -> Self {
+        Self {
+            inner: stream,
+            ended: false,
+        }
+    }
+
+    /// Get back the inner `Stream`.
+    pub fn into_inner(self) -> S {
+        self.inner
+    }
+}
+
+impl<T> Stream for StreamNotifyClose<T>
+where
+    T: Stream + Unpin,
```

Comment:
This `Unpin` bound is not necessary.
```suggestion
    T: Stream,
```
I'm guessing you added it because `self.ended = true` gave an error. You can fix that by doing `*self.project().ended = true` instead of adding `T: Unpin`.